iModelCore/iModelPlatform/DgnCore/linestyle/LsDb.cpp

```diff
@@ -631,9 +638,59 @@ DgnStyleId LineStyleElement::ImportLineStyle(DgnStyleId srcStyleId, DgnImportCon
         return DgnStyleId();
         }
 
-    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetAttributes(jsonObj));
+    BentleyStatus result = importer.GetDestinationDb().LineStyles().Insert(dstStyleId, name.c_str(), compId, LsDefinition::GetAttributes(jsonObj), LsDefinition::GetUnitDef(jsonObj));
     return (result == BSISUCCESS) ? dstStyleId : DgnStyleId();
     }
 
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DgnElementPtr LineStyleElement::_CloneForImport(DgnDbStatus* status, DgnModelR destModel, DgnImportContext& context) const
+    {
+    DgnDbStatus _status;
+    if (nullptr == status)
+        status = &_status;
+
+
+    DgnElementPtr newElem = T_Super::_CloneForImport(status, destModel, context);
+    if (!newElem.IsValid() || (DgnDbStatus::Success != *status))
+        return newElem;
+
+    if(!context.IsBetweenDbs())
+        return newElem;
+
+    DgnStyleId dstStyleId = QueryId(context.GetDestinationDb(), this->GetName().c_str());
+    if (dstStyleId.IsValid())
+        {
+        //  *** TBD: Check if the line style definitions match. If not, rename and remap
+        context.AddLineStyleId(DgnStyleId(this->GetElementId().GetValue()), dstStyleId);
+        return const_cast<DgnElement*>(context.GetDestinationDb().Elements().GetElement(dstStyleId).get());

```

Comment:
Can't return an element, remember caller is going to insert the element that is returned...

Calling context.AddLineStyleId is sufficient for geometric elements to remap the style, but should probably call context.AddElementId with this same pair of ids too so that generic element id remap lookup finds it too.

Then you need to return an invalid element to stop DgnElement::Import from doing the insert...

Reply:
Is returning `DgnElementPtr()` sufficient or should it just be nullptr? Also based on `DgnElements::InsertElement()` I noticed
`// don't allow elements that already have an id unless the "preassignedId" flag is set`
`if (element.m_elementId.IsValid() && !element.m_flags.m_preassignedId)`
I assume its still safer to just return an empty element in _CloneForImport?

Reply:
nullptr is fine. I just think it's cleaner to bail as soon as possible and not rely on the id check on insert...